### PR TITLE
Improvements to rest highlighting

### PR DESCRIPTION
**Pull request by amadeus**
* Added higher specificity to the dots in the spread declaration
* Linked the dots to `Noise`

### CODE REVIEW

**Comment by davidchambers** [comment]
> The proper term should be spread, not rest

```javascript
myFunction(...iterableObj)  // spread
```

```javascript
function(a, b, ...theArgs) {  // rest
  // ...
}
```

To which of the parenthesized sections above does `jsFuncArgs` apply?

**Comment by amadeus** [comment]
You're correct, I misread the spec, I'll rename them back, fix up this PR shortly.

**Comment by amadeus** [comment]
Fixed.

File: syntax/javascript.vim [hunk 1]
```diff
@@ -219,7 +219,8 @@ syntax match   jsGenerator      contained '\*' nextgroup=jsFuncName skipwhite
 syntax match   jsFuncName       contained /\<[a-zA-Z_$][0-9a-zA-Z_$]*/ nextgroup=jsFuncArgs skipwhite
 syntax region  jsFuncArgs       contained matchgroup=jsFuncParens start='(' end=')' contains=jsFuncArgCommas,jsFuncArgRest,jsAssignmentExpr,jsComment,jsLineComment nextgroup=jsFuncBlock keepend skipwhite skipempty
 syntax match   jsFuncArgCommas  contained ','
-syntax match   jsFuncArgRest    contained /\%(\.\.\.[a-zA-Z_$][0-9a-zA-Z_$]*\))/
+syntax match   jsFuncArgRest  contained /\%(\.\.\.[a-zA-Z_$][0-9a-zA-Z_$]*\))/ contains=jsFuncArgRestDots
```
Comment (davidchambers) [inline syntax/javascript.vim]: :arrow_right: :arrow_right:

Comment (amadeus) [inline syntax/javascript.vim]: Sorry, not sure what you mean?

Comment (amadeus) [inline syntax/javascript.vim]: Nevermind, you meant whitespace alignment, fixed

**Comment by davidchambers** [comment]
LGTM

**Pull request merged**